From david-b@pacbell.net Wed Jan 17 11:03:49 2007
From: Håvard Skinnemoen <hskinnemoen@atmel.com>
Date: Wed, 17 Jan 2007 11:03:29 -0800
Subject: USB: list atmel husb2_udc gadget controller
To: Greg KH <greg@kroah.com>
Cc: Haavard Skinnemoen <hskinnemoen@atmel.com>
Message-ID: <200701171103.30546.david-b@pacbell.net>
Content-Disposition: inline

From: Håvard Skinnemoen <hskinnemoen@atmel.com>

This identifies the driver for the Atmel HUSB2 Device Controller,
as integrated into the first AVR32 chip, the AT32AP700.

From: Håvard Skinnemoen <hskinnemoen@atmel.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/gadget_chips.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- gregkh-2.6.orig/drivers/usb/gadget/gadget_chips.h
+++ gregkh-2.6/drivers/usb/gadget/gadget_chips.h
@@ -75,6 +75,12 @@
 #define	gadget_is_pxa27x(g)	0
 #endif
 
+#ifdef CONFIG_USB_GADGET_HUSB2DEV
+#define gadget_is_husb2dev(g)	!strcmp("husb2_udc", (g)->name)
+#else
+#define gadget_is_husb2dev(g)	0
+#endif
+
 #ifdef CONFIG_USB_GADGET_S3C2410
 #define gadget_is_s3c2410(g)    !strcmp("s3c2410_udc", (g)->name)
 #else
@@ -169,5 +175,7 @@ static inline int usb_gadget_controller_
 		return 0x16;
 	else if (gadget_is_mpc8272(gadget))
 		return 0x17;
+	else if (gadget_is_husb2dev(gadget))
+		return 0x18;
 	return -ENOENT;
 }
